kubetest2/pkg/build/build.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package build implements a common system for building kubernetes for deployers to use.
+package build
+
+import (
+	"fmt"
+	"go/build"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+const (
+	target = "bazel-release"
+)
+
+var (
+	// This will need changed to support other platforms.
+	tarballsToExtract = []string{
+		"kubernetes.tar.gz",
+		"kubernetes-test-linux-amd64.tar.gz",
+		"kubernetes-test-portable.tar.gz",
+		"kubernetes-client-linux-amd64.tar.gz",
+	}
+)
+
+// Build builds kubernetes with the bazel-release target
+func Build() error {
+	// TODO(RonWeber): This needs options
+	goPath := os.Getenv("GOPATH")
```

Comment:
Use k8sdir() below? Also probably allow user to specify a customized path?

Reply:
Used k8sdir